### PR TITLE
Update circleci config to publish coverage report to codecov

### PR DESCRIPTION
**Pull request by jeffmay**
Add codecov back using the more secure CircleCI Orb tool (rather than the unsafe bash uploader)

### CODE REVIEW

**Review by vinit-garg — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,12 +39,14 @@ jobs:
             # fallback to using the latest cache if no exact match is found
             - v1-dependencies-
 
-      - run: cat /dev/null | sbt 'Test / compile'
+      # Compile, run tests, and produce coverage reports!
+      - run: cat /dev/null | sbt clean coverage test coverageReport coverageAggregate shutdown
```
Comment (paul-a-kennedy-rally) [inline .circleci/config.yml]: If you want to close `stdin`, 

```suggestion
      - run: exec 0<&- && sbt clean coverage test coverageReport coverageAggregate shutdown
```

Comment (paul-a-kennedy-rally) [inline .circleci/config.yml]: Or you could redirect in an empty input:
```suggestion
      - run: sbt clean coverage test coverageReport coverageAggregate shutdown << /dev/null
```

Either one of these would stop another shell process from being started.

**Review by paul-a-kennedy-rally — COMMENTED** [review]
I would fix the shell usage if the build system supports it.  It would start one fewer shell process.

File: .circleci/config.yml (hunk 1 above)
Comment (jeffmay) [reply .circleci/config.yml:43]: Neat! I just copied this from the template they posted. I like your solution better, but I would kinda prefer to stick to the template unless there is a good reason to deviate. Is there any advantages to this method?

**Review by paul-a-kennedy-rally — APPROVED** [review]
LGTM

**Pull request merged**